programs/cleos/CMakeLists.txt

```diff
@@ -4,6 +4,13 @@ if( UNIX AND NOT APPLE )
   set(rt_library rt )
 endif()
 
+# See "Note: Special instructions for GCC 8" on https://github.com/CLIUtils/CLI11
+if( ${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU" AND
+        ${CMAKE_CXX_COMPILER_VERSION} VERSION_GREATER_EQUAL 8 AND
+        ${CMAKE_CXX_COMPILER_VERSION} VERSION_LESS 9 )
+    add_compile_definitions(CLI11_HAS_FILESYSTEM=0)
+endif()
```

Comment:
This won't protect against some configurations because the limitation isn't due to the compiler but rather the C++ library being used.

Consider a fresh Ubuntu 18.04 install,
```
$ sudo apt-get install build-essential clang-8
$ echo "#include <filesystem>" > fs.cpp
$ clang++-8 -c --std=c++17 fs.cpp -o fs.o
fs.cpp:1:10: fatal error: 'filesystem' file not found
#include <filesystem>
         ^~~~~~~~~~~~
1 error generated.
```
In this case it fails even with clang 8 because the default c++ library on 18.04 is from gcc 7. (also, said configuration builds the develop branch fine today, so I don't consider something like this too academic)

As my previous comment mentioned, it probably actually is possible to get it working on gcc 7 by directing to experimental instead. And one day when we decide to replace boost::filesystem with std::filesystem we will need to handle all these permutations. But for now since we're not using this functionality is there any harm in keeping this simple and setting `CLI11_HAS_FILESYSTEM=0` across the board?

Reply:
I think we should just set that in main.cpp. @spoonincode do you see any benefit in having it hard-coded in cmake instead of main.cpp?

Reply:
I think it's fine either way.

Reply:
@spoonincode, the `if` condition in CMakeLists.txt also checks for the compiler vendor (GNU) before the version. That condition would not be true for clang++. Moreover, the CLI11 source also uses `__has_include` for c++17 before attempting to `#include <filesystem>`.

Reply:
If `CLI11_HAS_FILESYSTEM=0` works on all platforms, I would prefer having it on all platforms to avoid the differences.

Reply:
Sure, I'll define it for all platforms. Do you have a preference regarding CMakeLists.txt or main.cpp? I'd prefer to move it to main.cpp since it's no longer a configurable option.

Reply:
I like main.cpp as well.